src/class/Poll/Poll.cpp

```diff
@@ -126,27 +137,29 @@ void Poll::addWrite(int fd, t_handler h, Param p)
 	_params.push_back(p);
 }
 
-void Poll::_pop_index(int fd_index)
+void Poll::_pop_index(std::pair< int, ret_stt > p)
 {
-	int fd_to_close = _fds[fd_index].fd;
+	int fd_to_close = _fds[p.first].fd;
 	int count = 0;
 
 	for (size_t i = 0; i < _fds.size(); ++i)
 	{
 		if (_fds[i].fd == fd_to_close)
 			++count;
 	}
-	if (count == 1)
+	if (count == 1 && (p.second == DONE || p.second == ERROR))
```

Comment:
now Poll will check the reason we want to remove the fd.
the bug was that clientRead was DONE and removed cleintFD (and then Poll closed it) before we even started calling the cgi and then clinet write again.